### DISCUSSION TITLE
Enforcing custom naming convention with plugin

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14932] [Help Wanted!]
I’d like to use a naming convention for devices, where the name consists of the Device Role plus a 3 digit number that auto increments starting from 001 within each site.
For example, I have device roles ASW, LR, PDU, TV. The first TV in the site should be named TV001. With hundreds of devices, it will be difficult to know what the next number is, so I want Netbox to pick the number automatically.

I read through the plugin guide, and it seems like this should be pretty simple. I think I just need to use signals.py to receive the `pre_save` signal, check if the name=None then assign the right name after counting the existing devices.

So when I create a new device, I should always leave the name blank and then the plugin will name it correctly.

Does this sound like it would work, or does anybody see any red flags?

**Comment by rodvand** [comment]
Have you looked at the new Event Rules feature in NetBox 3.7? On creation of a device you can trigger a custom script which enforces this logic. I'm looking at this for automatically creating tags based on device attributes. 

I think it might be a simpler way to do it than creating a plugin.

**Reply by llamafilm** [reply]
Thanks, that's a great idea.  I implemented this and it's working great.  The only problem is if you click the newly created device's name in the green box (bottom right) very quickly after creation, the Device page still shows the old name until you refresh.  For me it takes about 2 seconds for the script to finish on my small test instance.  I'm not sure if that would change on a busier instance, but it's not a big deal.
